models/cice/ice_postprocessing_mod.f90

```diff
@@ -0,0 +1,949 @@
+    sice004  = max(0.0_r8, sice004)  ! salinities must be non-negative
+    sice005  = max(0.0_r8, sice005)  ! salinities must be non-negative
+    sice006  = max(0.0_r8, sice006)  ! salinities must be non-negative
+    sice007  = max(0.0_r8, sice007)  ! salinities must be non-negative
+    sice008  = max(0.0_r8, sice008)  ! salinities must be non-negative
+    qice001  = min(0.0_r8, qice001)  ! enthalpies (ice) must be non-positive
+    qice002  = min(0.0_r8, qice002)  ! enthalpies (ice) must be non-positive
+    qice003  = min(0.0_r8, qice003)  ! enthalpies (ice) must be non-positive
+    qice004  = min(0.0_r8, qice004)  ! enthalpies (ice) must be non-positive
+    qice005  = min(0.0_r8, qice005)  ! enthalpies (ice) must be non-positive
+    qice006  = min(0.0_r8, qice006)  ! enthalpies (ice) must be non-positive
+    qice007  = min(0.0_r8, qice007)  ! enthalpies (ice) must be non-positive
+    qice008  = min(0.0_r8, qice008)  ! enthalpies (ice) must be non-positive
+    qsno001  = min(0.0_r8, qsno001)  ! enthalphies (snow) must be non-positive
+    qsno002  = min(0.0_r8, qsno002)  ! enthalphies (snow) must be non-positive
+    qsno003  = min(0.0_r8, qsno003)  ! enthalphies (snow) must be non-positive    ! aicen = min(1.0_r8,aicen)  ! concentrations must not exceed 1 
+    Tsfcn = min(Tsmelt,Tsfcn)  ! ice/snow surface must not exceed melting
+ 
+    ! calculate aice, which might be negative or >1 at this point
+    vice = vicen(:,:,1)
+    do n = 2, Ncat  
+       vice = vice+vicen(:,:,n)
+    enddo
+ 
+    ! set negative aicen to zero
+    vicen = max(0.0_r8,vicen)   ! same for volumes (ice)
+ 
+    ! reclaculate aice, now it should be non-negative
+    vice_temp = vicen(:,:,1)
+    do n = 2, Ncat
+       vice_temp = vice_temp + vicen(:,:,n)
+    enddo
+ 
+    ! if vice <0, then set every category to 0
+    do j = 1, Ny
+       do i = 1, Nx
+          if (vice(i,j)<0._r8) then
+             vicen(i,j,:) = 0._r8
+          endif
+       enddo
+    enddo
+ 
+    ! shift negative post-adjustment volume values 
+    do n=1, Ncat
+       do j=1, Ny
+          do i=1, Nx
+             if (vice_temp(i,j) > 0._r8 .and. vice(i,j)>0._r8) then
+                vicen(i,j,n) = vicen(i,j,n) - (vice_temp(i,j)-vice(i,j))*vicen(i,j,n)/vice_temp(i,j)
+             endif  
+          enddo
+       enddo
+    enddo
+ 
+    ! calculate orignal caterogy thickness values
+    aicen_temp = aicen_original
+    where(aicen_temp==0) aicen_temp = -999
+ 
+    do n=1,Ncat
+       do j=1,Ny
+          do i=1,Nx
+             hicen_original(i,j,n) = vicen_original(i,j,n)/aicen_temp(i,j,n)
+             hsnon_original(i,j,n) = vsnon_original(i,j,n)/aicen_temp(i,j,n)
+          end do
+       end do
+    end do
+ 
+    where(hicen_original < 0)  hicen_original = 0.0_r8
+    where(hsnon_original < 0)  hsnon_original = 0.0_r8
+   
+    ! calculate the area implied by original category thickness and updated volume
+    aicen = vicen/hicen_original
```

Comment:
This is a divide by zero if hicen_original == 0.0_r8
and 

Should aicen = 0.0 if hicen_original = 0.0?
where(hicen_original/=0) aicen = vicen/hicen_original